lib/seccomp-tools/asm/compiler.rb

```diff
@@ -129,10 +139,17 @@ def compile_cmp(op, val, jt, jf)
       end
 
       def label_offset(label)
+        if label.is_a?(Integer)
+          raise ArgumentError, "Loop detected!" if label == 0
```

Comment:
label equals 0 stands for 'next line' but not itself.
And maybe we don't really need to distinguish 'loop' and 'backward jumping'?
I think 'backward jumping' is enough to remind users to prevent loop as well.

Reply:
whoops my bad on this one... and I think backwards jumping is good enough of an error